openflow13/multipart.go

```diff
@@ -57,28 +64,45 @@ func (s *MultipartRequest) UnmarshalBinary(data []byte) error {
 	n += 2
 	n += 4 // for padding
 
-	var req util.Message
-	switch s.Type {
-	case MultipartType_Aggregate:
-		req = s.Body.(*AggregateStatsRequest)
-		err = req.UnmarshalBinary(data[n:])
-	case MultipartType_Desc:
-		break
-	case MultipartType_Flow:
-		req = s.Body.(*FlowStatsRequest)
-		err = req.UnmarshalBinary(data[n:])
-	case MultipartType_Port:
-		req = s.Body.(*PortStatsRequest)
-		err = req.UnmarshalBinary(data[n:])
-	case MultipartType_Table:
-		break
-	case MultipartType_Queue:
-		req = s.Body.(*QueueStatsRequest)
-		err = req.UnmarshalBinary(data[n:])
-	case MultipartType_Experimenter:
-		break
+	if n < s.Header.Length {
+		var req util.Message
+		switch s.Type {
+		case MultipartType_Aggregate:
+			req = new(AggregateStatsRequest)
+			err = req.UnmarshalBinary(data[n:])
+		case MultipartType_Desc:
+			break
+		case MultipartType_Flow:
+			req = new(FlowStatsRequest)
+			err = req.UnmarshalBinary(data[n:])
+		case MultipartType_Port:
+			req = new(PortStatsRequest)
+			err = req.UnmarshalBinary(data[n:])
+		case MultipartType_Table:
+			break
```

Comment:
just wanted to confirm that this will trigger the `unsupported MultipartRequest type` error below and this is the expected behavior?